### PR TITLE
[SET] Add border & progress transition example

### PR DESCRIPTION
**Pull request by kacperkapusciak**
## Summary

This PR adds an code example that's using the new features available in Shared Element Transitions since React Native Reanimated 3.4.0

1. Animating border radius
2. Progress transition on iOS

### iOS

https://github.com/software-mansion/react-native-reanimated/assets/39658211/c5308669-4609-4796-82aa-2951adbeb5d2

### Android

https://github.com/software-mansion/react-native-reanimated/assets/39658211/3d3532cb-1705-4a6e-8a61-33d5308e9f75

### CODE REVIEW

File: app/src/examples/index.ts [hunk 1]
```diff
@@ -312,6 +313,11 @@ export const EXAMPLES: Record<string, Example> = {
     title: 'Volume slider & sensor',
     screen: VolumeExample,
   },
+  TransitionProgressExample: {
+    icon: '☕',
+    title: 'Progress transition example',
+    screen: ProgressTransitionExample,
```
Comment (tomekzaw) [inline app/src/examples/index.ts:319]: `TransitionProgress` or `ProgressTransition`?

File: app/src/examples/SharedElementTransitions/ProgressTransition.tsx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  createNativeStackNavigator,
+  NativeStackScreenProps,
+} from '@react-navigation/native-stack';
+import Animated, {
+  FadeInDown,
+  FadeInLeft,
+  FadeInUp,
+} from 'react-native-reanimated';
+
+import { useNavigation } from '@react-navigation/native';
```
Comment (tomekzaw) [inline app/src/examples/SharedElementTransitions/ProgressTransition.tsx]: never used

File: app/src/examples/SharedElementTransitions/ProgressTransition.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+// Image by <a href="https://www.freepik.com/free-photo/coffee-cup-wooden-boards-arrangement_29301195.htm#&position=27&from_view=collections">Freepik</a>
+
+const windowWidth = Dimensions.get('window').width;
+
+type StackParamList = {
+  Home: undefined;
+  Details: { tag: string };
+};
+
+const data = [
```
Comment (tomekzaw) [inline app/src/examples/SharedElementTransitions/ProgressTransition.tsx:44]:
```suggestion
const DATA = [
```

File: app/src/examples/SharedElementTransitions/ProgressTransition.tsx [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+  },
+];
+
+const Stack = createNativeStackNavigator<StackParamList>();
+
+function HomeScreen({
+  navigation,
+}: NativeStackScreenProps<StackParamList, 'Home'>) {
+  return (
+    <Animated.View style={[homeStyles.background, { paddingTop: 64 }]}>
```
Comment (tomekzaw) [inline app/src/examples/SharedElementTransitions/ProgressTransition.tsx:74]: plz don't use inline styles

File: app/src/examples/index.ts (hunk 1 above)
Comment (kacperkapusciak) [reply app/src/examples/index.ts:319]: There's no consistency on that matter in the codebase

image
image

@piaskowyk to which naming should we stick to?

File: app/src/examples/SharedElementTransitions/ProgressTransition.tsx (hunk 4 above)
Comment (kacperkapusciak) [reply app/src/examples/SharedElementTransitions/ProgressTransition.tsx]: Fixed in 03154e2

File: app/src/examples/SharedElementTransitions/ProgressTransition.tsx (hunk 3 above)
Comment (kacperkapusciak) [reply app/src/examples/SharedElementTransitions/ProgressTransition.tsx]: Fixed in 03154e2

File: app/src/examples/index.ts (hunk 1 above)
Comment (piaskowyk) [reply app/src/examples/index.ts:319]: `TransitionProgressEvent` - this is originally event name comes from `rn-screens`

`ProgressTransition` - this is shared element transition bases on progres value that comes from screen's `TransitionProgressEvent`

In summary we should stick with `ProgressTransition` I think.

Comment (kacperkapusciak) [reply app/src/examples/index.ts:319]: Alright, I'll try to keep the "progress transition" name consistent when creating docs for this in the future as well

Fixed in 9b42b9e

**Review by piaskowyk — APPROVED** [review]
👏
Well done!

**Pull request merged**